### DISCUSSION TITLE
can't get SenderUserID

### DISCUSSION THREAD

**Discussion by tretm** [Arman92/go-tdlib#60] [Q&A]
I updated the tdlib and it lacks in the message type two fields SenderUserID and SenderChatID instead of this there is a Sender field how can I get the SenderUserID in the message filtering function?

**Comment by motylkov** [comment]
I faced this trouble too.
In version 1.7.0, there are some changes regarding this:

Added support for messages sent on behalf of chats instead of users:
- Added the class MessageSender, representing a user or a chat which sent a message.
- Added the class MessageSenders, representing a list of message senders.
- Replaced the field sender_user_id with the field sender of the type MessageSender in the classes message and notificationTypeNewPushMessage.
- Added the class messageForwardOriginChat, which describe a chat as the original sender of a message.
- Added the ability to search messages sent by a chat by replacing the parameter sender_user_id with the parameter sender of the type MessageSender in the method searchChatMessages.
- Added the ability to specify a chat as a local message sender by replacing the parameter sender_user_id with the parameter sender of the type MessageSender in the method addLocalMessage.

**Comment by tretm** [comment]
Вы добавили user ID  в структуру sender? 


Пятница,  8 января 2021, 18:08 +03:00 от motylkov <notifications@github.com>:
>I faced this trouble too.
>In version 1.7.0, there are some changes regarding this:
>Added support for messages sent on behalf of chats instead of users:
>*  Added the class MessageSender, representing a user or a chat which sent a message.
>*  Added the class MessageSenders, representing a list of message senders.
>*  Replaced the field sender_user_id with the field sender of the type MessageSender in the classes message and notificationTypeNewPushMessage.
>*  Added the class messageForwardOriginChat, which describe a chat as the original sender of a message.
>*  Added the ability to search messages sent by a chat by replacing the parameter sender_user_id with the parameter sender of the type MessageSender in the method searchChatMessages.
>*  Added the ability to specify a chat as a local message sender by replacing the parameter sender_user_id with the parameter sender of the type MessageSender in the method addLocalMessage.
>—
>You are receiving this because you authored the thread.
>Reply to this email directly,  view it on GitHub , or  unsubscribe .

-- 
Михаил Третьяков

**Comment by Arman92** [comment]
I'll update the repo with the latest tdlib schema, it should fix it
